DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -929,3 +949,48 @@ int muon::sharedSegments( const reco::Muon& mu, const reco::Muon& mu2, unsigned
   
     return ret; 
 }
+
+void muon::setCutBasedSelectorFlags(reco::Muon& muon, 
+				    const reco::Vertex* vertex,
+				    bool run2016_hip_mitigation)
+{
+  // https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideMuonIdRun2
+  unsigned int selection = muon.getSelectionMask();
+  // Compute Id and Isolation variables
+  double chIso  = muon.pfIsolationR04().sumChargedHadronPt;
+  double nIso   = muon.pfIsolationR04().sumNeutralHadronEt;
+  double phoIso = muon.pfIsolationR04().sumPhotonEt;
+  double puIso  = muon.pfIsolationR04().sumPUPt;
+  double dbCorrectedIsolation = chIso + std::max( nIso + phoIso - .5*puIso, 0. ) ;
+  double dbCorectedRelIso = dbCorrectedIsolation/muon.pt();
+  double tkRelIso = muon.isolationR03().sumPt/muon.pt();
```

Comment:
all original values are floats. Is there a need to go to double precision here?

Reply:
Is there any harm? Impressed that you have noticed that, but I don't see any compiler warnings. These variables are used in calculations and it's better to use double precision for any calculations, so sooner or later translation from float to double will happen. See no reason to change that.

Reply:
the harm is in more CPU cycles.
I don't believe that there is any gain in precision in this context.
I guess the argument is that the CPU loss is tiny and you wrote it this way.